### ISSUE TITLE
[False-Positive] `Callable parameter usage violates definition` with preg_replace and str_replace

### ISSUE THREAD

**Issue by fraserredmond** [kalessil/phpinspectionsea#1144]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Issue type     | Bug/False-Positive                         |
| Plugin         | Php Inspections (EA Extended) 3.0.9.1 |
| Language level | PHP 7.2.2                        |

I'm not exactly sure, but I think it's only been happening the last week or two (I update the plugin within a day or two of new releases - when PhpStorm notifies me.)

https://github.com/kalessil/phpinspectionsea/issues/399 from last year seems similar, but the example code from that issue doesn't have any warnings.

Current behaviour (description/screenshot:)
--
Seems like the return value of `str_replace` and `preg_replace` alternate between `string` and `array`, but not always... I haven't been able to work out the pattern.

Below is an example. Other places in my code a set of `str_replace`s like that would be highlighted. The warnings go away if you type-cast (`= (string) str_replace`) every second line.

![image](https://user-images.githubusercontent.com/1796533/50175914-eabcde00-02f5-11e9-9baa-77863f969bda.png)

```
function foo (string $barStr) :string {
    $barStr = preg_replace("|a|", 'b', $barStr);
    $barStr = preg_replace("|b|", 'c', $barStr);
    $barStr = preg_replace("|c|", 'd', $barStr);

    $barStr = str_replace('a', 'b', $barStr);
    $barStr = str_replace('b', 'c', $barStr);
    $barStr = str_replace('c', 'd', $barStr);

    return $barStr;
}
```

Expected behaviour
--
No warnings in the above example function.

Environment details
---
PhpStorm 2018.3.1
Build #PS-183.4588.67, built on December 5, 2018
JRE: 1.8.0_152-release-1343-b16 amd64
JVM: OpenJDK 64-Bit Server VM by JetBrains s.r.o
Windows 10 10.0

**Comment by kalessil** [comment]
Thank you for reporting @fraserredmond, looks like a duplicate of #1129
